src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1417 @@
+    The Kubelet then pulls necessary container images and the container runtime create and starts containers. At this step
+    the Pod is said to be in "Running" state after a successful initialization.
+* **Container Startup Sequence**: If the init containers have been provided, they will run first and only after that the
+    pod main container will start. They are containers that are intended to be started, run initializing works and stops
+    before the actual container that should execute the application to deploy is started. If liveness and readiness probes
+    are provided, pod will become available only after the startup probe is successful and will be considered alive as long
+    as the liveness probe will be successful. After they are started, the containers are made available for service
+    discovery and networking.
+
+* **Notification**: After the pods are created, Kubelet will reports Pod creation status to the API Server, and confirms
+    successful container deployment. After this the cluster state in the `ectd` key-value store database will be updated.
+
+##### Visualization of the Process
+Here's a sequential diagram to illustrate the workflow:
+![Pod lifecycle](images/kubernetes-pod-lifecycle-sequential-diagram.png)
+
+##### Example
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+ name: workload-api
+ labels:
+    type: api
+    language: scala
+    framework: play-framework
+    service: workload
+spec:
+ containers:
+   - name: workload-api
+     image: aidocking/pfpw:20.20.31
+     ports:
+       - containerPort: 9000
+```
+
+Explanations:
+* **API Version and Kind**: The first two lines specify the Kubernetes Pods API version, which is crucial for Kubernetes
+to understand the intended action and correctly process the Pod creation request.
+* **Metadata Section**: This part provides descriptive information about the Pod, including its name and labels. While
+labels don't affect the Pod's behavior currently, they become important when working with Kubernetes controllers.
+* **Specification Section(`spec`)**: The spec defines the Pod behavior.
+* **Container Definition(`spec.containers`)**: The container details include its name, the Docker image to use, and the
+specific command to run when the container starts up. It is advised to specify only one container in a pod, but Kubernetes
+allows multiple containers to be defined in a single Pod configuration. In this case, it's a Play framework app container
+with the port 9000 exposed. It should be noted that pod are not intended to be created individually in real-life scenarios.
+They will be created through deployments.
+
+Deploy: `kubectl apply -f workload-api.yaml`
+
+Response:
+```text
+kubectl apply -f workload-api.yaml
+pod/workload-api created
+```
+View: `kubectl get pods` or `kubectl get -f workload-api.yaml`
+
+Response:
+```text
+kubectl get pods
+NAME           READY   STATUS    RESTARTS   AGE
+workload-api   1/1     Running   0          3m53s
+```
```

Comment:
Insert a call to the GET endpoint of the app to show how the app responds.

Reply:
Done at the Ingress level